woocommerce/storefront
# Update build tools

This PR introduces changes to the build process, including using node for the build process instead of grunt, our standard set of linting tools, and a GitHub workflow that replaces the Travis Job.

- Adds our suite of tooling for code styling/linting and closes #1656
- Updates the license information which closes #1669
- Adds browserlist config instead of defining manually which closes #1497
- Removes grunt and uses npm instead for building which closes #1433
- Moving away from grunt has fixed #1662 

Note, several CSS and JS files were modified due to the linting checks (things like spacing and indentation).

**Testing Instructions**

1. Confirm tests have passed; we have linting checks and the e2e task running via GitHub Workflow
2. `composer install && npm install && npm run build`
3. Confirm the build is successful and a zip archive is built. You can compare this to the last release zip - I used a diff tool and there are only minor differences due to updated tools. There should be no dev or build files within the release archive. **There should be** css files in assets/css and min.js files in assets/js.
4. Install the release archive and smoke test visual to ensure there are not style regressions.
5. Smoke test the JS functionality, for example, go to a product page and scroll so the add to cart button is out of view. This will cause sticky cart to appear. Or in admin, if you go to customizer for the first time you see a working guided tour.

When this PR is approved, it should also get the ready to merge label, just like the Blocks repo.
